internal/xds/translator/route.go

```diff
@@ -221,27 +218,30 @@ func buildXdsStringMatcher(irMatch *ir.StringMatch) *matcherv3.StringMatcher {
 	return stringMatcher
 }
 
-func buildXdsRouteAction(httpRoute *ir.HTTPRoute) *routev3.RouteAction {
-	return &routev3.RouteAction{
-		ClusterSpecifier: &routev3.RouteAction_Cluster{
-			Cluster: httpRoute.Destination.Name,
-		},
-		IdleTimeout: idleTimeout(httpRoute),
+func buildXdsRouteAction(backendWeights *ir.BackendWeights) *routev3.RouteAction {
+	if backendWeights.Invalid == 0 {
```

Comment:
can we add a comment here saying we only use a weighted cluster when we have invalid weights